api/tests/unit/domain/usecases/create-user-and-reconcile-to-schooling-registration-from-external-user_test.js

```diff
@@ -0,0 +1,173 @@
+const { domainBuilder, expect, sinon } = require('../../../test-helper');
+const createUserAndReconcileToSchoolingRegistrationFromExternalUser = require('../../../../lib/domain/usecases/create-user-and-reconcile-to-schooling-registration-from-external-user');
+
+describe('Unit | UseCase | create-user-and-reconcile-to-schooling-registration-from-external-user', function() {
+
+  let obfuscationService;
+  let tokenService;
+  let userReconciliationService;
+  let userService;
+  let authenticationMethodRepository;
+  let campaignRepository;
+  let userRepository;
+  let schoolingRegistrationRepository;
+  let studentRepository;
+
+  beforeEach(function() {
+    campaignRepository = {
+      getByCode: sinon.stub(),
+    };
+    tokenService = {
+      extractExternalUserFromIdToken: sinon.stub(),
+      createAccessTokenFromExternalUser: sinon.stub(),
+    };
+    userReconciliationService = {
+      findMatchingSchoolingRegistrationIdForGivenOrganizationIdAndUser: sinon.stub(),
+      checkIfStudentHasAnAlreadyReconciledAccount: sinon.stub(),
+    };
+    userRepository = {
+      getBySamlId: sinon.stub(),
+      get: sinon.stub(),
+      updateLastLoggedAt: sinon.stub(),
+    };
+    userService = {
+      createAndReconcileUserToSchoolingRegistration: sinon.stub(),
+    };
+  });
+
+  context('when user has saml id', function() {
+
+    it('should save last loggin date', async function() {
```

Comment:
J'ai un doute 

```suggestion
    it('should save last login date', async function() {
```
Il y en a un autre dans le context ou il n'a pas de saml ID